lib_test/test_fs_unix.ml

```diff
@@ -106,9 +108,22 @@ let read_zero_bytes () =
       sure %s is present in the test filesystem" content_file)
   | `Error e -> assert_fail e
 
+(* reasonable responses are `Ok [] or an error *)
+let read_negative_bytes () =
+  connect_or_fail () >>= fun fs ->
+  FS_impl.read fs content_file 0 (-5) >>= function
+  | `Ok [] -> Lwt.return_unit
+  | `Ok bufs ->
+    OUnit.assert_failure "reading -5 bytes from a non-empty file returned some cstructs"
+  | `Error (`No_directory_entry _) ->
+    OUnit.assert_failure (Printf.sprintf "read failed for a present file; please make
+      sure %s is present in the test filesystem" content_file)
+  | `Error (`Unknown_error _) -> Lwt.return_unit
```

Comment:
Yes, unfortunately OCaml doesn't make handling non-int integers very easy. I think there is agreement that throwing an exception (`Invalid_argument`) is fine in this case, since a negative value indicates a programming error elsewhere, not an expected runtime failure that needs to be handled.